torch/distributed/_shard/checkpoint/metadata.py

```diff
@@ -5,52 +5,36 @@
 import torch
 from torch.distributed._shard.sharded_tensor import (
     ShardedTensor,
-    ShardedTensorMetadata,
-    ShardMetadata,
 )
-
-TENSOR_TYPE = Union[torch.Tensor, ShardedTensor]
-STATE_DICT_TYPE = Dict[str, Any]
+from torch.distributed._shard.sharded_tensor.metadata import TensorProperties
 
 @dataclass
-class ShardStorageMetadata:
-    shard_metadata: ShardMetadata
-    # storage key used for this particular Shard
-    storage_key: str
-
-
-# Metadata for each param.
-@dataclass
-class ShardedTensorStorageMetadata:
-    # Metadata for the sharded tensor itself
-    tensor_metadata: ShardedTensorMetadata
-
-    # Storage info for each Shard. There's no ordering requirement for this list.
-    storage_metadata: List[ShardStorageMetadata]
-
+class ChunkStorageMetadata:
+    """
+    Each chunk is expected to have the same properties of the TensorStorageMetadata that includes it.
+    """
+    offsets: torch.Size
+    sizes: torch.Size
 
 @dataclass
 class TensorStorageMetadata:
-    # Storage key used for this tensor
-    storage_key: str
-
-    # Tensor sizes
+    properties: TensorProperties
     size: torch.Size
+    chunks: List[ChunkStorageMetadata]
 
 @dataclass
 class BytesStorageMetadata:
```

Comment:
is there still a need for `BytesStorageMetadata` or we just leave it here for completeness and future usage?

Reply:
We need it as a marker type and probable future usage.

Reply:
The issue is that we need to have an entry on the metadata state_dict for a given ByteIO entry to signal that it was saved to begin with.

Part of the representational issue here, which might be addressed later is that we're saving opaque stuff. One thing I expect this to be used with is down the line once we support non-pickle payloads that have a schema - and the schema would be here.